qulice-pmd/src/main/resources/com/qulice/pmd/ruleset.xml

```diff
@@ -196,4 +196,38 @@
         </properties>
         <priority>3</priority>
     </rule>
+
+    <rule name="ProhibitPublicStaticMethods"
+          message="Public static methods are prohibited."
+          language="java"
+          class="net.sourceforge.pmd.lang.rule.XPathRule">
+        <description>
+            Public static methods are prohibited.
+        </description>
+        <properties>
+            <property name="xpath">
+                <value>
+                    <![CDATA[
+                        //ClassOrInterfaceBodyDeclaration[
+                            MethodDeclaration[@Static='true' and @Public='true'
+                              and not (
+                               MethodDeclarator[
+                                count(FormalParameters/FormalParameter)=1
+                                and @Image='main'
+                                and FormalParameters/FormalParameter[1]/Type/ReferenceType/ClassOrInterfaceType[@Image='String']
+                                and FormalParameters/FormalParameter[@Varargs='true']
```

Comment:
@vkuchyn Valid signatures of main method are also:
`public static void main(String[] args)`, and `public static void main(String args[])`